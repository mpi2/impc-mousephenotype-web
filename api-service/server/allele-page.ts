import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { AlleleSummary } from "@/models";
const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchAlleleSummary(
  mgiGeneAccessionId: string,
  alleleName: string,
): Promise<AlleleSummary> {
  const endpointURL = `http://impc-allele-service.${KUBERNETES_NAMESPACE}:8080/v1/alleles?mgiGeneAccessionId=${mgiGeneAccessionId}&alleleName=${alleleName}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<AlleleSummary>(
        `/api/v1/alleles/${mgiGeneAccessionId}/${alleleName}`,
      )
    : fetchURL<AlleleSummary>(endpointURL));
}
