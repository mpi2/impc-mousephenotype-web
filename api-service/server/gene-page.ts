import { GeneOrder, GenePhenotypeHits, GeneSummary } from "@/models/gene";
import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { processGeneOrderResponse } from "@/hooks/gene-order.query";
import { processGenePhenotypeHitsResponse } from "@/hooks/significant-phenotypes.query";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchGeneSummary(
  mgiGeneAccessionId: string,
): Promise<GeneSummary> {
  const endpointURL = `http://impc-summary-service.${KUBERNETES_NAMESPACE}:8080/v1/gene_summary?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<GeneSummary>(
        `/api/v1/genes/${mgiGeneAccessionId}/summary`,
      )
    : fetchURL<GeneSummary>(endpointURL));
}

export async function fetchGenePhenotypeHits(
  mgiGeneAccessionId: string,
): Promise<Array<GenePhenotypeHits>> {
  const endpointURL = `http://impc-phenotype-hits-service.${KUBERNETES_NAMESPACE}:8080/v1/phenotypehits?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GenePhenotypeHits>>(
        `/api/v1/genes/${mgiGeneAccessionId}/phenotype-hits`,
      )
    : fetchURL<Array<GenePhenotypeHits>>(endpointURL));
  return processGenePhenotypeHitsResponse(data);
}

export async function fetchGeneOrderData(
  mgiGeneAccessionId: string,
): Promise<Array<GeneOrder>> {
  const endpointURL = `http://impc-order-service.${KUBERNETES_NAMESPACE}:8080/v1/orders?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GeneOrder>>(
        `/api/v1/genes/${mgiGeneAccessionId}/order`,
      )
    : fetchURL<Array<GeneOrder>>(endpointURL));
  return processGeneOrderResponse(data);
}
