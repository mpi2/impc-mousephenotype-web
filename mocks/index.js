export async function initMocks() {
  if (typeof window === "undefined") {
    const { server } = await import("./server");
    server.listen();
  } else {
    const { worker } = await import("./browser");
    return worker.start({
      serviceWorker: {
        url: "/data/mockServiceWorker.js",
        options: {
          scope: "/",
        }
      }
    });
  }
}

