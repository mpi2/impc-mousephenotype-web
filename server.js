const { createServer } = require('http');
const { parse } = require('url');
const next = require('next');

const dev = process.env.NODE_ENV !== 'production';
const hostname = 'localhost';
const port = 3000;
const app = next({ dev, hostname, port });
const handle = app.getRequestHandler();

app.prepare()
.then(() => {
    createServer(async (req, res) => {
        try {
            const parsedUrl = parse(req.url, true);
            const { pathname, query} = parsedUrl;
            const host = req.headers.host;
            console.log({ host, pathname });
            res.setHeader('X-IMPC-TEST', "true");
            if (host.includes('nginx.mousephenotype-dev.org')) {
                const modifiedPathName = pathname.replace("/data/", "");
                console.log({ modifiedPathName });
                await app.render(req, res, modifiedPathName, query);
                res.setHeader('X-PATH', JSON.stringify({ host, pathname, modifiedPathName }));
            } else {
                await handle(req, res, parsedUrl);
            }

        } catch (err) {
            console.error('Error occurred handling', req.url, err)
            res.statusCode = 500
            res.end('internal server error')
        }
    })
        .once('error', (err) => {
            console.error(err);
            process.exit(1);
        })
        .listen(port, () => {
            console.log(`Ready on http://${hostname}:${port}`);
        })
});