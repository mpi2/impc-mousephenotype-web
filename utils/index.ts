import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};
