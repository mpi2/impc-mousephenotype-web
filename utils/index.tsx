import React from "react";
import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};

export const formatPValue = (pValue: number) => {
  const pValueArray = Number.parseFloat(String(pValue))
    .toExponential(2)
    .split("e");
  return (
    <>
      {pValueArray[0]}x10<sup>{pValueArray[1].replace("+", "")}</sup>
    </>
  );
};

export const formatESCellName = (src: string) => {
  const [name, superscript, end] = src.split(/<|>/);
  return (
    <>
      {name}
      <sup>{superscript}</sup>
      {end}
    </>
  );
};

const fetchCache = {};
export { fetchCache };

export const toSentenceCase = (camelCase) => {
  if (camelCase) {
    const result = camelCase.replace(/([A-Z])/g, " $1");
    return result[0].toUpperCase() + result.substring(1).toLowerCase();
  }
  return "";
};
