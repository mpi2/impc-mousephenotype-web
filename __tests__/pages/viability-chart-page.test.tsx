import ViabilityChartPage from "@/app/supporting-data/viability/viability-chart-page";
import { screen, waitFor } from "@testing-library/react";
import { API_URL, renderWithClient, TEST_DATASETS_ENDPOINT } from "../utils";
import { testServer } from "../../mocks/server";
import { rest } from "msw";
import viabilityData from "../../mocks/data/tests/myo6-viability-data.json";
import datasetData from "../../mocks/data/tests/datasets/9deeb0258d5159af7911eebdc0ba2ed3.json";

window.ResizeObserver =
  window.ResizeObserver ||
  jest.fn().mockImplementation(() => ({
    disconnect: jest.fn(),
    observe: jest.fn(),
    unobserve: jest.fn(),
  }));

jest.mock("next/navigation", () => {
  const routerMock = {
    back: jest.fn(),
    push: jest.fn(),
  };
  return {
    useRouter: jest.fn().mockImplementation(() => routerMock),
    useSearchParams: jest
      .fn()
      .mockImplementation(
        () => new URLSearchParams({ mgiGeneAccessionId: "MGI:104785" }),
      ),
    usePathname: jest.fn(),
    useParams: jest.fn().mockImplementation(() => ({})),
  };
});

jest.mock("motion/react", () => {
  const FakeTransition = jest
    .fn()
    .mockImplementation(({ children }) => children);
  const FakeAnimatePresence = jest
    .fn()
    .mockImplementation(({ children }) => (
      <FakeTransition>{children}</FakeTransition>
    ));
  const motion = {
    tr: jest.fn().mockImplementation(({ children }) => <tr>{children}</tr>),
  };
  return {
    __esModule: true,
    motion,
    AnimatePresence: FakeAnimatePresence,
    default: jest.fn(),
  };
});

describe("Viability chart page", () => {
  it("renders correctly", async () => {
    testServer.use(
      rest.get(
        `${API_URL}/api/v1/genes/MGI:104785/dataset/viability`,
        (req, res, ctx) => {
          return res(ctx.json(viabilityData));
        },
      ),
      rest.get(
        `${TEST_DATASETS_ENDPOINT}/9deeb0258d5159af7911eebdc0ba2ed3.json`,
        (req, res, ctx) => {
          return res(ctx.json(datasetData));
        },
      ),
    );
    const { container } = renderWithClient(<ViabilityChartPage />);
    await waitFor(async () => {
      const rows = await screen.findAllByRole("table");
      return expect(rows.length).toEqual(2);
    });
    await waitFor(() =>
      expect(screen.getAllByRole("heading", { level: 1 })[0]).toHaveTextContent(
        "Viability data for Myo6 gene",
      ),
    );
    await waitFor(async () => {
      const rows = await screen.findAllByRole("row");
      return expect(rows.length).toEqual(12);
    });
    expect(container).toMatchSnapshot();
  });
});
