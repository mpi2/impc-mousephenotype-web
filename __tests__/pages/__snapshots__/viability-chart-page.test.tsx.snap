// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viability chart page renders correctly 1`] = `
<div>
  <div
    class="banner"
  >
    <div
      class="pb-4 pt-5 container container"
    >
      <div
        class="col-12 col-md-8 ps-4 pe-4"
      >
        <div
          class="tabs"
        >
          <a
            class="tab__active"
            href="/search"
          >
            Genes
          </a>
          <a
            class="tab"
            href="/search?type=pheno"
          >
            Phenotypes
          </a>
          <a
            class="tab"
            href="https://dev.mousephenotype.org/?s="
          >
            Help, news, blog
          </a>
        </div>
        <div
          class="inputCont"
        >
          <input
            class="input"
            placeholder="Search for a gene..."
            title="main search box"
            type="text"
            value=""
          />
          <button
            aria-describedby="svg-inline--fa-title-search-icon"
            class="searchBtn"
          >
            <svg
              aria-labelledby="svg-inline--fa-title-search-icon"
              class="svg-inline--fa fa-magnifying-glass "
              data-icon="magnifying-glass"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-search-icon"
              >
                Search button
              </title>
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="page container"
  >
    <div
      class="card"
    >
      <div
        class="subheading"
      >
        <span
          class="subheadingSection primary"
          data-testid="back-to-gene-page-link"
        >
          <a
            class="mb-3 link"
            href="/genes/MGI:104785"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-left "
              data-icon="arrow-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
                fill="currentColor"
              />
            </svg>
              Go Back to
             
            <i>
              <span
                style=""
              >
                Myo6
              </span>
            </i>
          </a>
        </span>
      </div>
      <h1
        class="mb-4 mt-2"
      >
        <strong
          class="text-capitalize"
        >
          Viability data for 
          <i>
            Myo6
          </i>
           gene
        </strong>
      </h1>
      <div
        class="mb-0"
      >
        <div
          style="display: flex; align-items: center; justify-content: space-between; flex-wrap: wrap; gap: 1rem;"
        >
          <span>
            5
             parameter / zygosity / metadata group combinations tested.
          </span>
        </div>
      </div>
      <div
        class="tableWrapper"
      >
        <table
          class="table striped data-comparison-table table table-bordered"
        >
          <thead>
            <tr>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: bold;"
                >
                  Allele
                   
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up sortIcon"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </th>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Viability
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Zygosity
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Phenotyping Centre
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Significant sex
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Life Stage
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Colony Id
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
            </tr>
            <tr />
          </thead>
          <tbody>
            <tr>
              <td>
                <i>
                  Myo6
                  <sup>
                    em1(IMPC)Tcp
                  </sup>
                </i>
              </td>
              <td>
                <strong>
                  Viable
                </strong>
              </td>
              <td>
                homozygote
              </td>
              <td>
                TCP
              </td>
              <td />
              <td>
                Early adult
              </td>
              <td>
                TCPR0947_ADUP
              </td>
            </tr>
            <tr>
              <td>
                <i>
                  Myo6
                  <sup>
                    em1(IMPC)Tcp
                  </sup>
                </i>
              </td>
              <td>
                <strong>
                  Viable
                </strong>
              </td>
              <td>
                homozygote
              </td>
              <td>
                TCP
              </td>
              <td />
              <td>
                Early adult
              </td>
              <td>
                TCPR0947_ADUP
              </td>
            </tr>
            <tr>
              <td>
                <i>
                  Myo6
                  <sup>
                    em1(IMPC)Tcp
                  </sup>
                </i>
              </td>
              <td>
                <strong>
                  Viable
                </strong>
              </td>
              <td>
                hemizygote
              </td>
              <td>
                TCP
              </td>
              <td />
              <td>
                Early adult
              </td>
              <td>
                TCPR0947_ADUP
              </td>
            </tr>
            <tr>
              <td>
                <i>
                  Myo6
                  <sup>
                    em1(IMPC)Tcp
                  </sup>
                </i>
              </td>
              <td>
                <strong>
                  Viable
                </strong>
              </td>
              <td>
                heterozygote
              </td>
              <td>
                TCP
              </td>
              <td />
              <td>
                Early adult
              </td>
              <td>
                TCPR0947_ADUP
              </td>
            </tr>
            <tr>
              <td>
                <i>
                  Myo6
                  <sup>
                    em1(IMPC)Tcp
                  </sup>
                </i>
              </td>
              <td>
                <strong>
                  Viable
                </strong>
              </td>
              <td>
                homozygote
              </td>
              <td>
                TCP
              </td>
              <td />
              <td>
                Early adult
              </td>
              <td>
                TCPR0947_ADUP
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        style="display: flex; justify-content: center;"
      />
    </div>
  </div>
  <div
    class="bg-grey pt-2"
    style="position: sticky; top: 0px; z-index: 100;"
  >
    <div
      class="container"
    >
      <div
        class="card"
      >
        <div>
          <div
            class="mb-4"
            style="display: flex; align-items: center; gap: 2rem;"
          >
            <h1
              style="margin: 0px;"
            >
              <strong>
                <i>
                  Myo6
                </i>
                 
                Homozygous males viability
                 data
              </strong>
            </h1>
            <span
              style="color: rgb(121, 118, 118); text-transform: uppercase; font-weight: 500; line-height: 1.2;"
            >
              Homozygous males viability
            </span>
          </div>
          <div
            class="fade alert alert-primary show"
            role="alert"
          >
            <p>
              Please note:
            </p>
            <ul>
              <li>
                data for different colonies will be presented separately (e.g. different alleles; same allele but different background strain; same allele but in different phenotyping centers)
              </li>
              <li>
                phenotype calls are made when a statistically significant abnormal phenotype is detected (that is, preweaning lethality or absence of expected number of homozygote pups based on Mendelian ratios)
              </li>
            </ul>
          </div>
        </div>
        <div
          class="details"
        >
          <div
            class="label"
          >
            Experiments performed
          </div>
          <div
            class="content"
          >
            <p>
              A 
              Viability Primary Screen
               phenotypic assay was performed on a mutant strain carrying the 
              <i>
                Myo6
                <sup>
                  em1(IMPC)Tcp
                </sup>
              </i>
               allele. The charts below show the proportion of wild type, heterozygous, and homozygous offspring.
            </p>
          </div>
          <div
            class="label"
          >
            Associated phenotype
          </div>
          <div
            class="content"
          >
            <strong>
              No significant association
            </strong>
          </div>
          <div
            class="label"
          >
            Zygosity
          </div>
          <div
            class="content"
          >
            <span>
              homozygote
            </span>
          </div>
          <div
            class="label"
          >
            Testing protocol
          </div>
          <div
            class="content"
          >
            <a
              class="link primary"
              href="https://www.mousephenotype.org/impress/ProcedureInfo?action=list&procID=1267&pipeID=9"
            >
              Viability Primary Screen
            </a>
          </div>
          <div
            class="label"
          >
            Measured value
          </div>
          <div
            class="content"
          >
            Homozygous males viability
          </div>
          <div
            class="label"
          >
            Testing environment
          </div>
          <div
            class="content"
          >
            <span
              class="primary link"
            >
              Lab conditions and equipment
            </span>
          </div>
          <div
            class="label"
          >
            Life stage
          </div>
          <div
            class="content"
          >
            <span>
              Early adult
            </span>
          </div>
          <div
            class="label"
          >
            Background Strain
          </div>
          <div
            class="content"
          >
            <span>
              involves: C57BL/6NCrl
            </span>
          </div>
          <div
            class="label"
          >
            Phenotyping center
          </div>
          <div
            class="content"
          >
            <span>
              TCP
            </span>
          </div>
          <div
            class="label"
          />
          <div
            class="content"
          >
            <p
              class="small"
            >
              * The high throughput nature of the IMPC means that large control sample sizes may accumulate over a long period of time. See the animal welfare guidelines for more information.
            </p>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-lg-4"
        >
          <div
            class="card"
          >
            <h2
              class="primary"
            >
              Total counts (male and female)
            </h2>
            <div
              class="chartWrapper"
            >
              <canvas
                height="0"
                role="img"
                style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
                width="0"
              />
            </div>
          </div>
        </div>
        <div
          class="col-lg-4"
        >
          <div
            class="card"
          >
            <h2
              class="primary"
            >
              Male counts
            </h2>
            <div
              class="chartWrapper"
            >
              <canvas
                height="0"
                role="img"
                style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
                width="0"
              />
            </div>
          </div>
        </div>
        <div
          class="col-lg-4"
        >
          <div
            class="card"
          >
            <h2
              class="primary"
            >
              Female counts
            </h2>
            <div
              class="chartWrapper"
            >
              <canvas
                height="0"
                role="img"
                style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
                width="0"
              />
            </div>
          </div>
        </div>
        <div
          class="col-lg-6"
        >
          <div
            class="card"
          >
            <div
              class="tableWrapper"
            >
              <table
                class="table striped table table-bordered"
              >
                <thead>
                  <tr>
                    <th
                      rowspan="2"
                      width="50%"
                    >
                      <button
                        class="inlineButton headerButton disabled"
                        style="font-weight: inherit;"
                      >
                        Sex
                         
                      </button>
                    </th>
                    <th
                      rowspan="2"
                      width="8.333333333333332%"
                    >
                      <button
                        class="inlineButton headerButton disabled"
                        style="font-weight: inherit;"
                      >
                        WT
                         
                      </button>
                    </th>
                    <th
                      rowspan="2"
                      width="8.333333333333332%"
                    >
                      <button
                        class="inlineButton headerButton disabled"
                        style="font-weight: inherit;"
                      >
                        Het
                         
                      </button>
                    </th>
                    <th
                      rowspan="2"
                      width="8.333333333333332%"
                    >
                      <button
                        class="inlineButton headerButton disabled"
                        style="font-weight: inherit;"
                      >
                        Hom
                         
                      </button>
                    </th>
                    <th
                      rowspan="2"
                      width="10%"
                    >
                      <button
                        class="inlineButton headerButton disabled"
                        style="font-weight: inherit;"
                      >
                        Hemi ¹
                         
                      </button>
                    </th>
                    <th
                      rowspan="2"
                      width="8.333333333333332%"
                    >
                      <button
                        class="inlineButton headerButton disabled"
                        style="font-weight: inherit;"
                      >
                        Total
                         
                      </button>
                    </th>
                  </tr>
                  <tr />
                </thead>
                <tbody>
                  <tr>
                    <td>
                      Male and female
                    </td>
                    <td>
                      38
                    </td>
                    <td>
                      95
                    </td>
                    <td>
                       
                      43
                    </td>
                    <td />
                    <td>
                      176
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Male
                    </td>
                    <td>
                      22
                    </td>
                    <td>
                      37
                    </td>
                    <td>
                      25
                    </td>
                    <td />
                    <td>
                      105
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Female
                    </td>
                    <td>
                      16
                    </td>
                    <td>
                      37
                    </td>
                    <td>
                      18
                    </td>
                    <td>
                      N/A
                    </td>
                    <td>
                      71
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          class="col-lg-6"
        >
          <div
            class="card"
          >
            <h2>
              Statistical analysis API access
            </h2>
            <p>
              <a
                class="link"
                href="https://www.ebi.ac.uk/mi/impc/solr/genotype-phenotype/select?q=*:*&rows=2147483647&sort=p_value+asc&wt=xml&fq=marker_accession_id:%22MGI:104785%22&fq=phenotyping_center:(%22TCP%22)&fq=allele_accession_id:%22MGI:6257724%22&fq=pipeline_stable_id:TCP_001&fq=parameter_stable_id:IMPC_VIA_063_001&fq=zygosity:homozygote&fq=strain_accession_id:%22MGI:2683688%22"
                target="_blank"
              >
                Genotype phenotype association
                 (
                XML
                ) 
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-up-right-from-square fa-xs "
                  data-icon="up-right-from-square"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M352 0c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9L370.7 96 201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L416 141.3l41.4 41.4c9.2 9.2 22.9 11.9 34.9 6.9s19.8-16.6 19.8-29.6V32c0-17.7-14.3-32-32-32H352zM80 32C35.8 32 0 67.8 0 112V432c0 44.2 35.8 80 80 80H400c44.2 0 80-35.8 80-80V320c0-17.7-14.3-32-32-32s-32 14.3-32 32V432c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V112c0-8.8 7.2-16 16-16H192c17.7 0 32-14.3 32-32s-14.3-32-32-32H80z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
